py/frequenz/client/electricity_trading/_types.py

```diff
@@ -0,0 +1,1220 @@
+    """Private and confidential trade, not visible in the public
+    order book and has no market impact. (Not yet supported)."""
+
+    @classmethod
+    def from_pb(
+        cls, order_type: electricity_trading_pb2.OrderType.ValueType
+    ) -> OrderType:
+        """Convert a protobuf OrderType value to OrderType enum.
+
+        Args:
+            order_type: Order type to convert.
+
+        Returns:
+            Enum value corresponding to the protobuf message.
+        """
+        if not any(e.value == order_type for e in cls):
+            _logger.warning("Unknown order type %s. Returning UNSPECIFIED.", order_type)
+            return cls.UNSPECIFIED
+
+        return cls(order_type)
+
+
+class MarketSide(enum.Enum):
+    """Which side of the market the order is on, either buying or selling."""
+
+    UNSPECIFIED = electricity_trading_pb2.MarketSide.MARKET_SIDE_UNSPECIFIED
+    """The side of the market has not been set."""
+
+    BUY = electricity_trading_pb2.MarketSide.MARKET_SIDE_BUY
+    """Order to purchase electricity, referred to as a 'bid' in the order book."""
+
+    SELL = electricity_trading_pb2.MarketSide.MARKET_SIDE_SELL
+    """Order to sell electricity, referred to as an 'ask' or 'offer' in the order book."""
+
+    @classmethod
+    def from_pb(
+        cls, market_side: electricity_trading_pb2.MarketSide.ValueType
+    ) -> MarketSide:
+        """Convert a protobuf MarketSide value to MarketSide enum.
+
+        Args:
+            market_side: Market side to convert.
+
+        Returns:
+            Enum value corresponding to the protobuf message.
+        """
+        if not any(e.value == market_side for e in cls):
+            _logger.warning(
+                "Unknown market side %s. Returning UNSPECIFIED.", market_side
+            )
+            return cls.UNSPECIFIED
+
+        return cls(market_side)
+
+
+class OrderState(enum.Enum):
+    """State of an order."""
+
+    UNSPECIFIED = electricity_trading_pb2.OrderState.ORDER_STATE_UNSPECIFIED
+    """The order state is not known. Usually the default state of a newly created order object
+    before any operations have been applied."""
+
+    PENDING = electricity_trading_pb2.OrderState.ORDER_STATE_PENDING
+    """The order has been sent to the marketplace but has not yet been confirmed. This can be due
+    to awaiting validation or system processing."""
+
+    ACTIVE = electricity_trading_pb2.OrderState.ORDER_STATE_ACTIVE
+    """The order has been confirmed and is open in the market. It may be unfilled or partially
+    filled."""
+
+    FILLED = electricity_trading_pb2.OrderState.ORDER_STATE_FILLED
+    """The order has been completely filled and there are no remaining quantities on the order."""
+
+    CANCELED = electricity_trading_pb2.OrderState.ORDER_STATE_CANCELED
+    """The order has been cancelled. This can occur due to a cancellation request by the market
+    participant, system, or market operator."""
```

Comment:
You want 1 L or 2?

```suggestion
    CANCELED = electricity_trading_pb2.OrderState.ORDER_STATE_CANCELED
    """The order has been canceled. This can occur due to a cancellation request by the market
    participant, system, or market operator."""
```
or
```suggestion
    CANCELLED = electricity_trading_pb2.OrderState.ORDER_STATE_CANCELLED
    """The order has been cancelled. This can occur due to a cancellation request by the market
    participant, system, or market operator."""
```

Reply:
(I guess you want 1 because that's what the protobuf definition has.)